test/e2e/storage/csi_mock/csi_volume_expansion.go

```diff
@@ -201,6 +201,131 @@ var _ = utils.SIGDescribe("CSI Mock volume expansion", func() {
 			})
 		}
 	})
+	ginkgo.Context("CSI online volume expansion with secret[Feature:CSINodeExpandSecret]", func() {
+		var stringSecret = map[string]string{
+			"username": "admin",
+			"password": "t0p-Secret",
+		}
+		trackedCalls := []string{
+			"NodeExpandVolume",
+		}
+		tests := []struct {
+			name          string
+			disableAttach bool
+			expectedCalls []csiCall
+
+			// Called for each NodeExpandVolume calls, with counter incremented atomically before
+			// the invocation (i.e first value will be 1).
+			nodeExpandHook func(counter int64) error
+		}{
+			{
+				name: "should expand volume without restarting pod if attach=on, nodeExpansion=on, csiNodeExpandSecret=on",
+				expectedCalls: []csiCall{
+					{expectedMethod: "NodeExpandVolume", expectedError: codes.OK, expectedSecret: stringSecret},
+				},
+			},
+		}
+		for _, t := range tests {
+			test := t
+			ginkgo.It(test.name, func(ctx context.Context) {
+				var (
+					err        error
+					hooks      *drivers.Hooks
+					secretName = "test-secret"
+					secret     = &v1.Secret{
+						ObjectMeta: metav1.ObjectMeta{
+							Namespace: f.Namespace.Name,
+							Name:      secretName,
+						},
+						StringData: stringSecret,
+					}
+				)
+				if test.nodeExpandHook != nil {
+					hooks = createPreHook("NodeExpandVolume", test.nodeExpandHook)
+				}
+				params := testParameters{enableResizing: true, enableNodeExpansion: true, enableCSINodeExpandSecret: true, hooks: hooks}
+				if test.disableAttach {
+					params.disableAttach = true
+					params.registerDriver = true
+				}
+
+				m.init(ctx, params)
+				ginkgo.DeferCleanup(m.cleanup)
+
+				if secret, err := m.cs.CoreV1().Secrets(f.Namespace.Name).Create(context.TODO(), secret, metav1.CreateOptions{}); err != nil {
+					framework.Failf("unable to create test secret %s: %v", secret.Name, err)
+				}
+
+				sc, pvc, pod := m.createPod(ctx, pvcReference)
+				gomega.Expect(pod).NotTo(gomega.BeNil(), "while creating pod for resizing")
+
+				if !*sc.AllowVolumeExpansion {
+					framework.Fail("failed creating sc with allowed expansion")
+				}
+				if sc.Parameters == nil {
+					framework.Fail("failed creating sc with secret")
+				}
+				if _, ok := sc.Parameters[csiNodeExpandSecretKey]; !ok {
+					framework.Failf("creating sc without %s", csiNodeExpandSecretKey)
+				}
+				if _, ok := sc.Parameters[csiNodeExpandSecretNamespaceKey]; !ok {
+					framework.Failf("creating sc without %s", csiNodeExpandSecretNamespaceKey)
+				}
+				err = e2epod.WaitForPodNameRunningInNamespace(ctx, m.cs, pod.Name, pod.Namespace)
+				framework.ExpectNoError(err, "Failed to start pod1: %v", err)
+
+				fmt.Printf("######pvc %v######\n", pvc)
```

Comment:
@zhucan please remove above debugging statement.

Reply:
@humblec  I'm a little confused， before added the debugging statement, the test result is failed, but after it, it's successful.